test/services/ValidationEngineSpec.scala

```diff
@@ -285,7 +286,16 @@ class ValidationEngineSpec  extends SpecBase with MockitoSugar {
         when(mockXmlValidationService.validateXml(any())).thenReturn((elem, missingAddressErrors))
 
         val expectedErrors = Seq(GenericError(20, "Enter a Street"))
-        validationEngine.validateFile(source, businessRulesCheckRequired = false) mustBe ValidationFailure(expectedErrors)
+        validationEngine.validateFile(source, businessRulesCheckRequired = false) mustBe Right(ValidationFailure(expectedErrors))
+      }
+
+      "must throw an exception if XML parser throws an exception (e.g. missing closing tags)" in new SetUp {
+        when(mockXmlValidationService.validateXml(any())).thenThrow(new RuntimeException)
+
+        an[Exception] mustBe thrownBy {
+          when(mockXmlValidationService.validateXml(any())).thenThrow(new RuntimeException)
+          validationEngine.validateFile(source)
+        }
```

Comment:
That's not the behavior you want validateXml doesnt throw an exception it wraps an exception in a Left